### ISSUE TITLE
Stuck at Loading Screen

### ISSUE THREAD

**Issue by dchokola** [QuantumPhysique/trale#65]
Upgraded to 0.6.1 through FDroid and was not able to get past the loading screen.

Unfortunately, I was not able to downgrade without first uninstalling and lost all my progress. It seems user data is not stored in a user folder which persists between installs.

Now, with no saved data, 0.6.1 works fine.

**Comment by braniii** [comment]
@dchokola thx for opening this issue. I am very sorry to hear that you lost your measurements :grimacing:
Downgrading is prohibited by Android, but next time (hopefully there is no next issue :see_no_evil: ) please open the issue before erasing your data. Then, we can compile you the old version with higher version code to allow reverting the change.

@gwosd was facing a similar issue when upgrading to v0.6.0. This is why we have withdrawn this version and pushed a fix in v0.6.1. 
Using a clean installation of v0.5.0 creating some measurements and then upgrading to v0.6.1 works fine for me. So I have no idea what went wrong. Did you change some settings, set a target weight, changed the unit? And how many measurements did you had?

**Comment by braniii** [comment]
And regarding the user data. I have to check how this can be implemented, so that it persists between installations. For now it is recommended to regularly backup the measurements via the settings.

**Comment by constantine-fanourakis** [comment]
@braniii I am in the same situation, but haven't uninstalled the app. I would like to keep my data if possible. As far as changes to the settings, I have changed the unit.

**Comment by braniii** [comment]
@constantine-fanourakis a version for downgrade is running at 5de43fabac0211fec3465f54dfe48ac4ec67c980. I will upload it here in 10min, once it is done.

**Comment by braniii** [comment]
[app-arm64-v8a-release.zip](https://github.com/QuantumPhysique/trale/files/14781239/app-arm64-v8a-release.zip)

Here, is the apk taken from https://github.com/QuantumPhysique/trale/actions/runs/8458590626. Sorry, needed to zip it because github does not allow to upload executable files. Can you please try to install it and tell me if it helped to downgrade?

**Comment by sphars** [comment]
> [app-arm64-v8a-release.zip](https://github.com/QuantumPhysique/trale/files/14781239/app-arm64-v8a-release.zip)

Thanks, I was able to install the app from this action and pull a backup.

I was not able import the backup into 0.6.1, but I was able to do so with the release app above (but only after reset option from within the app). There was a mild crash when changing the interpolation setting, had to switch back and forth. May make another issue for some minor things I've noticed.

**Comment by braniii** [comment]
> > [app-arm64-v8a-release.zip](https://github.com/QuantumPhysique/trale/files/14781239/app-arm64-v8a-release.zip)
> 
> Thanks, I was able to install the app from this action and pull a backup.
> 
> I was not able import the backup into 0.6.1, but I was able to do so with the release app above (but only after reset option from within the app). There was a mild crash when changing the interpolation setting, had to switch back and forth. May make another issue for some minor things I've noticed. Edit: the mild crash with interpolation is the same as reported in #25, so looks like that may still be an issue.

Hi, I am sorry for all the inconvenience. I am glad to hear that you manage to backup your data. The provided apk is simply version 0.5.0 with increased version number to 0.6.2 so to allow downgrades. Therefore, all fixed issues persist.
Whould it be possible to share your backup, so that we can find the bug? I guess you can open a new confidential issue.

**Comment by braniii** [comment]
@sphars if you do not want to share your private weight measurements, you can also simply create new measurements in the old version, backup and then upgrade. As long as this fails, it would be of great help to find the bug.

Which unit and interpolation are you using?

**Comment by braniii** [comment]
@dchokola @sphars @constantine-fanourakis I have played around a bit with the app and still didn't manage to reproduce the error. Could you please tell me which Android Version you are using?

**Comment by sphars** [comment]
Here's how I am able to reproduce the loading screen hang error on v0.6.1:

- Install 0.5.0 from F-Droid
- Open app and import backup. Confirm data is showing properly 
- Upgrade to version 0.6.1 via F-Droid 
- Open app
- See loading screen hang

If I proceed install 0.6.2 from the post above, with no other changes from my steps, the app loads again and data is showing as expected. I'll make a private issue with my backup data for you.

I've been using lbs originally, though my backup file is set to kg. Interpolation is the default option (3rd)

Pixel 8, Android 14

If it helps, I can try and capture a logcat when I have access to my PC later

**Comment by braniii** [comment]
@sphars thank you so much for your help!

**Comment by dchokola** [comment]
Seems @sphars has much more quality debugging data than me, but FWIW, I'm on Android 11.

**Comment by braniii** [comment]
I've tried to fix #70 and uploaded there an apk in my latest comment. Since I am still unable to reproduce this issue, it would be of great help if anyone with could check if it solves this issue here as well.

**Comment by sphars** [comment]
Downloaded the APK in [your comment](https://github.com/QuantumPhysique/trale/issues/70#issuecomment-2039469231) and did an upgrade from 0.5.0 to this one. It looks like it worked, both upgrading directly and with a clean install and import of data. The app does report its version is 0.6.2, but it looks right. So I think we're good now?

**Comment by braniii** [comment]
I've released a new version to github. So either I recommend to wait with upgrading from v0.5.0 until v0.6.2 is on fdroid, or updating from github release directly.

**Comment by damoasda** [comment]
I got v0.6.2 via fdroid and it works again for me, the issue is fixed.
